Script args at tail for more intuitive usage

atifaziz opened the pull request:
An attempt at addressing #11. I say attempt because while it worked for the script case, it didn't for the eval one & I ran out of my quantum. :)

filipw commented:
This is OK, I don't think `eval` doesn't need arguments IMHO, since the whole code is supplied from the outside anyway.

Can you please update the readme to reflect the arg changes?

atifaziz commented:
@filipw I've updated the usage in the README. I have also changed the usage so that `--` separates the args passed to `dotnet script` and those passed to the script itself. This has to be done because the command-line args processing library you're using didn't allow `dotnet script` and the script to overload the arguments. `RemainingArguments` would only contain those not processed by `dotnet script`. This means the script could not have its own help (`-h`) or use `-d` to mean something else.

When you're ready to merge (:shipit:), I can squash the changes to a single commit.

filipw commented:
thanks a lot!
